### PR TITLE
Disconnect sockets

### PR DESCRIPTION
**Pull request by bucko13**
This fixes a kind of bug where the SocketManager was not handling client disconnects properly and leaving event subscriptions even if no more clients were subscribed to those events. 

Expected behavior (illustrated in the new test) goes something like this:
- two clients send a request to the server to subscribe to a new event (say `block connect`) on a node `test`, responding with an event `foo bar`
- When a `block connect` is heard on the `test` client an event is sent to all clients subscribed to that with the responseEvent `foo bar`
- If one client unsubscribes or disconnects, the above behavior is maintained for the first client. `test` client still listens for `block connect` events
- If both clients are unsubscribed, `test` client no longer listens for `block connect` events.

Actual behavior (before this fix) was mostly the same as above except:
- disconnects were not being handled, clients had to explicitly call to `unsubscribe`
- When all subscribed clients were disconnected, or unsubscribed, the clients on the server were still listening for the events. So, for example, if you had one client that was first connecting to a bcoin node and requesting new txs or blocks, and then switched to a bcash or hsd node, the SocketManager would continue receiving events for new txs and blocks for the bcoin node even though no more clients are subscribed to those events.

This PR depends on #122

### CODE REVIEW

**Comment by tynes** [comment]
It seems like this should get rebased on top of https://github.com/bpanel-org/bpanel/pull/122?

**Comment by bucko13** [comment]
Yep! That's the plan. Just wanted to get this up for reference.

**Comment by bucko13** [comment]
Actually, to clarify, since we're squashing PR commits now, it doesn't really matter whether it gets rebased or not since the commit history gets lost. I'll rebase/merge with development once #122 gets in and it should make the diff look clearer. I just forked from that branch to get started on this while the other was in review.

**Comment by bucko13** [comment]
Alright, this is ready to go now as #122 has been merged to development branch.

File: configs/webpack.common.config.js [hunk 1]
```diff
@@ -39,6 +39,7 @@ module.exports = () => {
         hsd$: `${MODULES_DIR}/hsd/lib/hsd-browser`,
         bledger: `${MODULES_DIR}/bledger/lib/bledger-browser`,
         bmultisig: `${MODULES_DIR}/bmultisig/lib/bmultisig-browser`,
+        bcrytp: `${MODULES_DIR}/bcrypto`,
```
Comment (tynes) [inline configs/webpack.common.config.js]: typo

Comment (bucko13) [reply configs/webpack.common.config.js]: I have no idea what you're talking about 🙄

**Pull request merged**